lib/template-mixins.js

```diff
@@ -95,12 +95,13 @@ module.exports = function (t, fields, options) {
         });
     }
 
-    function optionGroup(key) {
-        return {
+    function optionGroup(key, extension) {
+        extension = extension || {};
+        return _.extend(extension, {
             'key': key,
             'error': this.errors && this.errors[key],
-            'legend': fields[key].legend && fields[key].legend.value,
-            'options': _.map(fields[key].options, function (obj) {
+            'legend': fields[key] && fields[key].legend && fields[key].legend.value,
```

Comment:
What was the behaviour here before? I'm not 100% sure that we don't want the template to error out if you attempt to reference an undefined field.